pkg/api/core/v1/models/models.go

```diff
@@ -90,10 +90,10 @@ type ApplicationStage struct {
 // ApplicationOrigin is the part of the manifest describing the origin of the application
 // (sources). At most one of the fields may be specified / not empty.
 type ApplicationOrigin struct {
-	Kind      int    `yaml:"-"` // Hidden from yaml. Type tag to simplify struct usage.
-	Container string `yaml:"container,omitempty"`
-	Git       GitRef `yaml:"git,omitempty"`
-	Path      string `yaml:"path,omitempty"`
+	Kind      int     `                           yaml:"-" ` // Hidden from json and yaml. Type tag to simplify struct usage.
```

Comment:
Is this indeed hidden from JSON without a `json:"-"` clause ?

Reply:
Ah, I left the comment. I had to remove the `json:"-"` because it was actually used to marshal/unmarshal the Kind.

I don't really like to expose the property as-is

```
{
    "Kind": 1,
    "path": "C:\\Documents\\app"
}
```

but change it in other ways would need a bit of refactor. What do you think?

The best would be to have something like an enum mapping the Kind `int`, so exposing something more explicit:

```
{
    "type": "ORIGIN_PATH",
    "path": "C:\\Documents\\app"
}
```

Reply:
That is weird wrt `json:"-"`. This is, AFAIK the proper way for excluding the field from (un)marshalling.

Looking at the tests you have I see that you have `Kind` in the JSON. That is what I am expecting for this tagging, without the `json:"-"` clause. Are you saying that when you add this clause the tests still pass, with `Kind` in the JSON ?

Go has no enums as we are used from C. See https://github.com/epinio/epinio/blob/5f3872d1e3c5eca11937f6020d459f8129502542/pkg/api/core/v1/models/models.go#L100-L105 for an example of a Go enum.

Reply:
No, I meant that originally I have added the clause but other tests were failing, because they were implicitly relying on the `Kind` field.
So I had to remove it and keep the marshaling as is (with the `Kind` exposed).

I know about the enum, unfortunately (is a feature that I actually miss). To add this kind of behaviour the only way would be to add a custom type/struct and implement the `Unmarshal` interface I suppose.
Probably overkill.

Reply:
Oh. I see. ... Were the failing tests API tests ? Because IIRC it is exposed there. Makes the handling on each side nicer (albeit opens each side up to getting structure where kind and actual actual information do not match).

We could drop it from the API, close that attack vector ... Not in this issue and PR.

Reply:
I recommend adding a note in the code about this issue with Kind not hidden.

Resolving as ok, will approve.